staging/src/k8s.io/kubectl/pkg/cmd/apply/apply.go

```diff
@@ -618,16 +619,20 @@ See https://kubernetes.io/docs/reference/using-api/server-side-apply/#conflicts`
 		if err != nil {
 			return err
 		}
-		patchBytes, patchedObject, err := patcher.Patch(info.Object, modified, info.Source, info.Namespace, info.Name, o.ErrOut)
+		patchedObject, err := patcher.Patch(info.Object, modified, info.Source, info.Namespace, info.Name, o.ErrOut)
 		if err != nil {
-			return cmdutil.AddSourceToErr(fmt.Sprintf("applying patch:\n%s\nto:\n%v\nfor:", patchBytes, info), info.Source, err)
+			return cmdutil.AddSourceToErr(fmt.Sprintf("applying patch:\n%v\nto:\n%v\nfor:", patchedObject, info), info.Source, err)
 		}
 
-		info.Refresh(patchedObject, true)
+		changed := false
```

Comment:
This is the condition I was referring to earlier, if we have that boolean information returned from `patcher.patch` we can safely drop this piece of code.